src/pygeoprocessing/geoprocessing.py

```diff
@@ -1630,8 +1630,18 @@ def reproject_vector(
     # Get the SR of the original_layer to use in transforming
     base_sr = layer.GetSpatialRef()
 
-    # Create a coordinate transformation
-    coord_trans = osr.CoordinateTransformation(base_sr, target_sr)
+    # Create a coordinate transformation [GDAL 2 NOT supported]
+    # In GDAL 3.0 spatial references have been updated. One difference is 
+    # how transforms work with lat/lon coordinate systems. Transforms
+    # expect (lat, lon) order in GDAL 3.0 as opposed to thge GIS friendly
+    # (lon, lat). Thus if using a (lat, lon) srs set for backwards 
+    # compatability. 
+    if not base_sr.IsProjected():
+        base_sr.SetAxisMappingStrategy(osr.OAMS_TRADITIONAL_GIS_ORDER)
+    if not target_sr.IsProjected():
+        target_sr.SetAxisMappingStrategy(osr.OAMS_TRADITIONAL_GIS_ORDER)
```

Comment:
If I understand the issue correctly, in this function everything is contained inside of `osr.Geometry` objects which should be internally consistent with whatever the axis order is for whatever the projection is. We shouldn't encounter an x vs. y vs lat vs long because it's not exposed at this level. Is that possible? Or did you make this change because of an issue in InVEST or something?

Reply:
p.s. I'm not totally sure about this, but I'd want to make sure we know why we're setting the axis mapping strategy. If there's a failed test or something that's obvious, otherwise I keep thinking "how do we know if it's lat/lng or x/y". In the bounding box routine below we really don't know. But here all the geometry should know and `ogr` should be internally consistent?

Reply:
RE: a thread we had on slack, yes this is necessary, but not necessary to have the `if` statements in there. And I have one other overall PR comment related to this I'll put up top.

Reply:
Yep, totally agree.

Reply:
Yep, totally agree.